### DISCUSSION TITLE
Equivalent new command for `convert.exe` from an older version of ImageMagick

### DISCUSSION THREAD

**Discussion by mkanet** [ImageMagick/ImageMagick#6193] [Help]
### ImageMagick version

6.9.2-0 Q16 x64 2015-08-15

### Operating system, version and so on

Windows 11

### Description

I've been using a very old version of ImageMagick `Convert.exe` command for many years to create an animated gif from a series of images.  When I upgraded to a new version of ImageMagick, I don't even see the `Convert.exe` command there anytime.

```
convert.exe -delay 100 -loop 0 $imagesArray.fullname C:\animated.gif
```

**If I want to do EXACTLY the same thing in the latest version of ImageMagick for Windows, what command do I type?**

Lastly, and most importantly, if I were to upgrade to the new version, does it take advantage of hardware acceleration from my 13th generation Intel i7 137000K CPU when executing this command?  Or, is it limited to just basic CPU?

**Comment by fmw42** [comment]
In IM 7, use magick in place of convert.  Other tools would be magick identify, magick montage, etc. 

`magick.exe  -delay 100 -loop 0 $imagesArray.fullname C:\animated.gif`

If you want to use convert, you have to install the legacy components when installing on Windows.

**Comment by snibgo** [comment]
Some recent pre-built binaries for Windows (eg v7.1.0-62) do not include OpenMP, so there is no multi-threading. I don't know why this is. A developer may comment.

They do include OpenCL, so a GPU may massively increase performance of some operations.

**Reply by mkanet** [reply]
Thank you.  I noticed OpenCL was included.  I would love to use a version of that has both OpenMP and OpenCL.  **I have 16 core 13th gen Intel CPU.**  Do you know which is the latest version that includes both?  I'm using Q16 64bit.

**Comment by snibgo** [comment]
The latest pre-built version that I have that includes OpenCL and OpenMP is:
```
C:\web\im>%IMG7104%magick -version
Version: ImageMagick 7.1.0-32 Q16-HDRI x64 a2b2c08:20220430 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193131107)
```
The next version I have is v7.1.0-42, which doesn't have OpenMP.

You can build IM yourself on Windows, eg using Cygwin tools. Including OpenMP is simple. I haven't figured out how to include OpenCL.

**Reply by mkanet** [reply]
Thank you.  Actually, I just checked, mine looks just like yours!  It probably explains why it was so fast, I almost thought it was broken. :)

```
 magick -version
Version: ImageMagick 7.1.1-4 Q16-HDRI x64 10ad43d:20230318 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193431942)
```
